core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -615,7 +616,7 @@ class Building : RulesetStatsObject(), INonPerpetualConstruction {
 
         for ((resource, amount) in getResourceRequirements())
             if (civInfo.getCivResourcesByName()[resource]!! < amount) {
-                rejectionReasons.add(RejectionReason.ConsumesResources.toInstance(resource.getConsumesAmountString(amount)))
+                rejectionReasons.add(RejectionReason.ConsumesResources.toInstance(resource.getNeedMoreAmountString(amount)))
```

Comment:
It looks like you need to change the argument of `getNeedMoreAmountString()` since `amount` is the number of resources you need for the building/unit but `civInfo.getCivResourcesByName()[resource]` is how much the player currently has. If you're trying to build a nuke that requires 2 uranium but only have 1 spare uranium, the text would say "Need 2 more uranium" when you only need 1 more.